lib/Parse/ParseDecl.cpp

```diff
@@ -1211,45 +1255,61 @@ Parser::parseStorageRestrictionsAttribute(SourceLoc AtLoc, SourceLoc Loc) {
       hasNextProperty = consumeIf(tok::comma);
     } while (hasNextProperty);
 
-    return false;
+    return status;
   };
 
-  auto parseArgument = [&](bool isOptional = false) -> bool {
+  auto parseArgument = [&](bool isOptional = false) -> ParserStatus {
+    ParserStatus status;
+
     if (Tok.is(tok::r_paren) && isOptional)
-      return false;
+      return status;
 
     Identifier accessLabel;
     SourceLoc loc;
     parseOptionalArgumentLabel(accessLabel, loc);
 
     if (accessLabel.empty()) {
-      diagnose(Loc, diag::missing_storage_restrictions_attr_label);
-      return true;
+      if (consumeIf(tok::code_complete)) {
+        if (this->CodeCompletionCallbacks) {
+          this->CodeCompletionCallbacks->completeDeclAttrParam(
+              CustomSyntaxAttributeKind::StorageRestrictions, /*Index*/ 0,
```

Comment:
To avoid confusion, should this use `StorageRestrictionsCompletionKind::Label` instead of index `0`?

Reply:
Good catch 👍🏽